### PR TITLE
sink(ticdc): do not return error when close the sink

### LINKED ISSUE #8195: simplify the sink close by not return error
as title

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #8195 

### What is changed and how it works?

when close the sink, no need to return the error, this can simplify the close logic and caller error handling, also make sure all components is fully closed.

#### Tests 
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by zhaoxinyu — APPROVED** [review]

File: cdc/sinkv2/ddlsink/mysql/mysql_ddl_sink.go [hunk 1]
```diff
@@ -204,13 +204,11 @@ func (m *mysqlDDLSink) WriteCheckpointTs(_ context.Context, _ uint64, _ []*model
 }
 
 // Close closes the database connection.
-func (m *mysqlDDLSink) Close() error {
-	if err := m.db.Close(); err != nil {
-		return errors.Trace(err)
+func (m *mysqlDDLSink) Close() {
+	if m.db != nil {
+		m.db.Close()
```
[the remark this answers is not in the document]
Comment (3AceShowHand) [reply cdc/sinkv2/ddlsink/mysql/mysql_ddl_sink.go]: add log if close the `db` meet any error

**Comment by 3AceShowHand** [comment]
> In practice, a close without error means that it may not have an io operation. A close with error in general means that there is an underlying "io operation". I'm not sure this change will cause misunderstandings. Also this is not consistent with our other error handling, which I would prefer the processor manager printing warnings originally. This way we can make sure that errors are printed in most cases, without each underlying implementation making sure it prints the correct warning log.

I agree that we should always show that there is any error happening. 

Originally, we throw the erorr to the caller, and the caller print the warn log, I think this is complicated, because we have to handle such non-critical error in multiple methods.

By removing the error from the close, and printing the log if mee any error immediately, the caller's burden is reduced.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**